prompt_poet/template_loaders.py

```diff
@@ -0,0 +1,74 @@
+"""A Prompt Poet (PP) template registry."""
+
+import os
+import jinja2 as j2
+
+
+class TemplateLoader:
+    """Base class for template loaders.
+
+    This abstract class defines the interface for loading templates from various sources.
+    Concrete implementations should handle specific template loading scenarios like
+    local filesystem, remote filesystem, etc.
+    """
+
+    def load(self) -> j2.Template:
+        """Load and return a Jinja2 template from the source."""
+        pass
+
+    def id(self) -> str:
+        """Generate a unique identifier for this template loader configuration."""
+        pass
+
+
+def parse_template_path(template_path: str) -> tuple[str, str]:
+    """Parse the template path to determine the template source."""
+    template_dir, template_name = os.path.split(template_path)
+    if not template_dir:
+        template_dir = "."
+    return template_dir, template_name
```

Comment:
is this intended to be used outside of this module? if not can you make it private `_parse_template_path`? alternatively include this on the TemplateLoader and make it overridable?

Reply:
i was thinking about using it in other loader (like GCSLoader), but yes, it could be private here as implementation outside of this library can be different. And this implementation is simple enough.